cloudformation/stack_topic_handler.js

```diff
@@ -41,162 +58,255 @@ async function handleBudgetAlert(event, context) {
-      UsePreviousTemplate: true,
-      Parameters: newParams,
-      Capabilities: ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"]
+    StackName: stackName,
+    UsePreviousTemplate: true,
+    Parameters: newParams,
+    Capabilities: ["CAPABILITY_IAM", "CAPABILITY_NAMED_IAM"],
   });
 }
 
 async function handleASGMessage(message, context) {
   const asgName = message.AutoScalingGroupName;
   const asgEvent = message.Event;
-  const region = "${AWS::Region}";
-  const recordName = "${LowerStackName.Value}-app.${InternalZoneInfo.Name}.";
-  const hostedZoneId = "${InternalZoneInfo.Id}";
+  const REGION = "${AWS::Region}";
+  const RECORD_NAME = "${LowerStackName.Value}-app.${InternalZoneInfo.Name}.";
+  const HOSTED_ZONE_ID = "${InternalZoneInfo.Id}";
   const ttl = 15;
+  const DOMAIN_URL = "${InternalZoneInfo.Name}.";
 
-  //console.log(asgEvent);
-
-  if (asgEvent === "autoscaling:EC2_INSTANCE_LAUNCH" || asgEvent === "autoscaling:EC2_INSTANCE_TERMINATE" || asgEvent === "INSTANCE_REBOOT") {
-    //console.log("Handling Launch/Terminate Event for " + asgName);
+  if (
+    asgEvent === "autoscaling:EC2_INSTANCE_LAUNCH" ||
+    asgEvent === "autoscaling:EC2_INSTANCE_TERMINATE" ||
+    asgEvent === "INSTANCE_REBOOT"
+  ) {
+    // console.log('Handling Launch/Terminate Event for ' + asgName)
 
-    const autoscaling = new AWS.AutoScaling({region});
-    const ec2 = new AWS.EC2({region});
+    const autoscaling = new AWS.AutoScaling({ region: REGION });
+    const ec2 = new AWS.EC2({ region: REGION });
     const route53 = new AWS.Route53();
 
-    const asgResponse = await promisify(autoscaling.describeAutoScalingGroups.bind(autoscaling))({
+    const asgResponse = await promisify(
+      autoscaling.describeAutoScalingGroups.bind(autoscaling)
+    )({
       AutoScalingGroupNames: [asgName],
-      MaxRecords: 1
+      MaxRecords: 1,
     });
 
-    const recordSets = (await promisify(route53.listResourceRecordSets.bind(route53))({
-      StartRecordName: recordName,
-      StartRecordType: "A",
-      HostedZoneId: hostedZoneId,
-      MaxItems: "100"
-    })).ResourceRecordSets;
+    const recordSets = (
+      await promisify(route53.listResourceRecordSets.bind(route53))({
+        HostedZoneId: HOSTED_ZONE_ID,
+        MaxItems: "100",
+      })
+    ).ResourceRecordSets;
+
+    const instanceIds = asgResponse.AutoScalingGroups[0].Instances.map(
+      (i) => i.InstanceId
+    );
+
+    // Sometimes instanceIds is an empty array
+    // Will cause an error on the ec2 describe instances
+    if (!instanceIds.length) {
+      console.log(message.EC2InstanceId);
+      instanceIds.push(message.EC2InstanceId);
+    }
 
-    const instanceIds = asgResponse.AutoScalingGroups[0].Instances.map(i => i.InstanceId);
     const instanceInfo = await promisify(ec2.describeInstances.bind(ec2))({
       DryRun: false,
-      InstanceIds: instanceIds
+      InstanceIds: instanceIds, // Can conflict with other ec2 instances. Only get the ones in the asg.
```

Comment:
If you don't understand it, then it's better to remove it. 

I deleted this line and attempted to include all EC2 instances but ran into IP address conflicts across ASGs and it created a "Bad Gateway" error when too many multivalue answer routing policies were kept on the hosted zone.

Reply:
Ah, that context helps. I found the comment confusing because it just says what the code is doing -- filtering to a set of desired instance Ids, so It was odd that you mentioned it could conflict, since I assume instance ids are unique across all ASGs. Makes more sense that the IP addresses could conflict.